File: conferencia_presenca/presenca.go
package main

import (
	"bufio"
	"encoding/json"
	"fmt"
	"net/http"
	"os"
	"strings"
)

type Commit struct {
	Commit struct {
		Message string `json:"message"`
		Date    string `json:"committer"`
	} `json:"commit"`
}

func main() {
	token := os.Getenv("GITHUB_TOKEN")
	if token == "" {
		panic("GITHUB_TOKEN not set")
	}
	aula := os.Getenv("AULA")
	if token == "" {
		panic("AULA not set")
	}

	file, err := os.Open("repos.txt")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)

	fmt.Println("Alunas presentes: ")
	for scanner.Scan() {
		line := scanner.Text()
		name, repoURL, _ := strings.Cut(line, " | ")
		parts := strings.Split(repoURL, "github.com/")
		if len(parts) < 2 {
			fmt.Println(name, " - Erro ao dividir url do repositório em partes")
			continue
		}
		repoPath := parts[1]
		apiURL := fmt.Sprintf("https://api.github.com/repos/%s/commits", repoPath)

		req, _ := http.NewRequest("GET", apiURL, nil)
		req.Header.Set("Authorization", "Bearer "+token)
		client := &http.Client{}
		resp, err := client.Do(req)
		if err != nil {
			fmt.Println(name, " - Erro na requisição")
			continue
		}
		defer resp.Body.Close()

		var commits []Commit
		json.NewDecoder(resp.Body).Decode(&commits)

		found := false
		for _, c := range commits {
			if strings.Contains(c.Commit.Message, fmt.Sprint("aula ", aula, " -")) {
				// fmt.Printf("[✅] %s commitou: %s\n", name, c.Commit.Message)
				fmt.Printf("%s\t\t%d\n", name, 1)
				found = true
				break
			}
		}
		if !found {
			fmt.Printf("%s\t\t%d\n", name, 0)
		}
	}
}

File: conferencia_presenca/presenca.js
const fs = require("fs");
const https = require("https");

const token = process.env.GITHUB_TOKEN;
const aula = process.env.AULA;

if (!token) {
  console.error("Erro: GITHUB_TOKEN não definido.");
  process.exit(1);
}

if (!aula) {
  console.error("Erro: AULA não definida.");
  process.exit(1);
}

const reposFile = "repos.txt";

if (!fs.existsSync(reposFile)) {
  console.error("Arquivo repos.txt não encontrado.");
  process.exit(1);
}

const lines = fs.readFileSync(reposFile, "utf-8").split("\n").filter(Boolean);

console.log("Chamada:");

lines.forEach((line) => {
  const [name, repoURL] = line.split(" | ");

  const parts = repoURL.split("github.com/");
  if (parts.length < 2) {
    console.log(`${name} - Erro ao dividir URL do repositório`);
    return;
  }

  const repoPath = parts[1];
  const apiURL = `https://api.github.com/repos/${repoPath}/commits`;

  const options = {
    headers: {
      "User-Agent": "presenca-script",
      Authorization: `Bearer ${token}`,
    },
  };

  https.get(apiURL, options, (res) => {
    let data = "";

    res.on("data", (chunk) => {
      data += chunk;
    });

    res.on("end", () => {
        try {
        const commits = JSON.parse(data);
        if (!Array.isArray(commits)) {
            console.log(`${name}\t\t0`);
          return;
        }            
        const found = commits.some((commit) =>
          commit.commit.message.includes(`aula ${aula} -`)
        );

        console.log(`${name}\t\t${found ? 1 : 0}`);
      } catch (err) {
        // console.log(`${name} - Erro ao processar resposta da API: ${err.message}`);
      }
    });
  }).on("error", (err) => {
    console.log(`${name} - Erro na requisição`);
  });
});
